sourcegraph/sourcegraph
# svelte: Add infinity scroll to commit page and simplify history panel implementation

This commit adds infinity scroll behavior to the commit page, building on top of [Apollo's pagination features.](https://www.apollographql.com/docs/react/pagination/overview). This has the advantage of not having to introduce custom logic in the component for handling pagination, which makes the components simpler. Now they only need to work with a data list and a function to fetch more data. **Update:** Using the Apollo cache for this is not as straightforward as it seemed. See [my follow up comment](https://github.com/sourcegraph/sourcegraph/pull/59639#issuecomment-1897416068)

Additionally [SvelteKit's snapshot feature](https://kit.svelte.dev/docs/snapshots) is used to track the position of the scrollbar and the expanded/collapsed state of individual file diffs.

A drawback is that the field policy code is not very strongly typed.

I applied the same technique to the history panel which simplifies its implementation a lot.


## Test plan

- Scroll on commit page. A loading indicator is shown when more diffs are being loaded and replaced by the new data.
- Scroll history panel, same as above. Additionally, when switching between files the previous list of commits should be removed and a loading indicator should be shown instead.
